File: market/test/java/ru/yandex/market/replenishment/autoorder/service/CompetitorPriceLoaderTest.java
package ru.yandex.market.replenishment.autoorder.service;

import java.time.LocalDateTime;
import java.util.Map;

import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ActiveProfiles;

import ru.yandex.market.common.test.db.DbUnitDataSet;
import ru.yandex.market.replenishment.autoorder.config.FunctionalTest;
import ru.yandex.market.replenishment.autoorder.service.yt.loader.CompetitorPriceLoader;
import ru.yandex.market.yql_test.annotation.YqlTest;

import static ru.yandex.market.replenishment.autoorder.utils.TestUtils.setMockedYtTableExistsCheckService;
@ActiveProfiles("unittest")
public class CompetitorPriceLoaderTest extends FunctionalTest {
    @Autowired
    CompetitorPriceLoader loader;

    @Test
    @YqlTest(
            schemasDir = "/yt/schemas",
            schemas = {
                    "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test/dco_upload_table" +
                            "/2021-06-18T08--colon--00--colon--00"
            },
            csv = "CompetitorPriceLoaderTest_import.yql.csv",
            yqlMock = "CompetitorPriceLoaderTest_import.yql.mock.json"
    )
    @DbUnitDataSet(after = "CompetitorPriceLoaderTest.after.csv")
    public void importTest() {
        setMockedObjects(true);
        loader.load();
    }

    @Test
    @YqlTest(
            schemasDir = "/yt/schemas",
            schemas = {
                    "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test/dco_upload_table" +
                            "/2021-06-18T04--colon--00--colon--00",
                    "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test/dco_upload_table" +
                            "/2021-06-18T08--colon--00--colon--00"
            },
            csv = "CompetitorPriceLoaderTest_tableNotExists.yql.csv",
            yqlMock = "CompetitorPriceLoaderTest_tableNotExists.yql.mock.json"
    )
    @DbUnitDataSet(after = "CompetitorPriceLoaderTest.after.csv")
    public void whenTableNotExistUsePrevious() {
        setMockedObjects(false);
        loader.load();
    }

    private void setMockedObjects(boolean currentPathExists) {
        setExistYtTablesChecker(currentPathExists);
        setTestTime(LocalDateTime.of(2021, 6, 18, 10, 0));
    }

    private void setExistYtTablesChecker(boolean currentPathExists) {
        var path = "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test/dco_upload_table" +
                "/2021-06-18T08:00:00";
        var pathBefore = "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test" +
                "/dco_upload_table/2021-06-18T04:00:00";
        setMockedYtTableExistsCheckService(loader,
                Map.of(path, currentPathExists,
                        pathBefore, true));


//        var yt = mock(Yt.class);
//        var cypress = mock(Cypress.class);
//        var ytFactory = Mockito.mock(YtFactory.class);
//        when(ytFactory.getYt(YtCluster.HAHN)).thenReturn(yt);
//        when(yt.cypress()).thenReturn(cypress);
//        when(cypress.exists(YPath.simple(path))).thenReturn(currentPathExists);
//        when(cypress.exists(YPath.simple(pathBefore))).thenReturn(true);
//        ReflectionTestUtils.setField(loader, "ytFactory", ytFactory);

    }

}

File: market/test/java/ru/yandex/market/pricingmgmt/loaders/competitorPrice/CompetitorPriceLoaderTest.kt
package ru.yandex.market.pricingmgmt.loaders.competitorPrice

import org.junit.jupiter.api.Test
import org.mockito.ArgumentMatchers.any
import org.mockito.Mockito
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.test.mock.mockito.MockBean
import ru.yandex.market.common.test.db.DbUnitDataSet
import ru.yandex.market.pricingmgmt.api.ControllerTest
import ru.yandex.market.pricingmgmt.loaders.CompetitorPriceLoader
import ru.yandex.market.pricingmgmt.service.TimeService
import ru.yandex.market.yql_test.annotation.YqlTest
import ru.yandex.market.yt.util.table.YtTableService
import java.time.LocalDateTime

class CompetitorPriceLoaderTest : ControllerTest() {

    @Autowired
    lateinit var loader: CompetitorPriceLoader

    @MockBean
    lateinit var timeService: TimeService

    @MockBean
    @Qualifier("ytHahnTableService")
    lateinit var ytHahnTableService: YtTableService

    @Test
    @DbUnitDataSet(after = ["CompetitorPriceLoaderTest_import.after.csv"])
    @YqlTest(
        schemasDir = "/yt/schemas",
        schemas = [
            "//home/market/production/mstat/dictionaries/shop_real_supplier/latest",
            "//home/market/production/ir/ultra-controller/supplier_to_market_sku",
            "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test/dco_upload_table/2022-02-01T16--colon--00--colon--00"
        ],
        csv = "CompetitorPriceLoaderTest_import.yql.csv",
        yqlMock = "CompetitorPriceLoaderTest.yql.mock"
    )
    fun import() {
        Mockito.`when`(ytHahnTableService.isTableExists(any())).thenReturn(true)
        Mockito.`when`(timeService.getNowDateTime()).thenReturn(LocalDateTime.of(2022, 2, 1, 16, 0))
        loader.load()
    }

    @Test
    @DbUnitDataSet(after = ["CompetitorPriceLoaderTest_import.after.csv"])
    @YqlTest(
        schemasDir = "/yt/schemas",
        schemas = [
            "//home/market/production/mstat/dictionaries/shop_real_supplier/latest",
            "//home/market/production/ir/ultra-controller/supplier_to_market_sku",
            "//home/market/production/mstat/analyst/regular/mbi/blue_prices/test/dco_upload_table/2022-02-01T16--colon--00--colon--00"
        ],
        csv = "CompetitorPriceLoaderTest_import.yql.csv",
        yqlMock = "CompetitorPriceLoaderTest.yql.mock"
    )
    fun import_whenTableNotExistUsePrevious() {
        Mockito.`when`(ytHahnTableService.isTableExists(any())).thenReturn(false)
        Mockito.`when`(timeService.getNowDateTime()).thenReturn(LocalDateTime.of(2022, 2, 1, 20, 0))
        loader.load()
    }
}
